### PR TITLE
Use public keys for the mock mix message encoding

### PR DESCRIPTION
**Pull request by youngjoon-lee**
## 1. What does this PR implement?

- Add the trait `MixMessage` and implement `MockMixMessage`.
- Include public keys for the `MockMixMessage`
- Add the struct `Membership`, load it from node/test configurations, and use it for establishing connections and constructing mix messages.

## 2. Does the code have enough context to be clearly understood?

Ja

## 3. Who are the specification authors and who is accountable for this PR?

@youngjoon-lee @AlejandroCabeza @danielSanchezQ 

## 4. Is the specification accurate and complete?
Ja

## 5. Does the implementation introduce changes in the specification?
Nein

## Checklist

> [!WARNING]  
> Do not merge the PR if any of the following is missing:

* [ ] 1. Description added.
* [ ] 2. Context and links to Specification document(s) added.
* [ ] 3. Main contact(s) (developers and specification authors) added
* [ ] 4. Implementation and Specification are 100% in sync including changes. This is critical.
* [ ] 5. Link PR to a specific milestone.

### CODE REVIEW

**Review by AlejandroCabeza — APPROVED** [review]

File: nomos-mix/core/src/message_blend/mod.rs [hunk 1]
```diff
@@ -86,13 +114,31 @@ where
...
 
-pub trait MessageBlendExt: Stream<Item = Vec<u8>> {
-    fn blend(self, message_blend_settings: MessageBlendSettings) -> MessageBlendStream<Self>
+pub trait MessageBlendExt<M>: Stream<Item = Vec<u8>>
+where
+    M: MixMessage,
+    M::PrivateKey: Clone + Serialize + DeserializeOwned + PartialEq,
+    M::PublicKey: Clone + Serialize + DeserializeOwned + PartialEq,
+{
+    fn blend<R>(
```
[the remark this answers is not in the document]
Comment (youngjoon-lee) [reply nomos-mix/core/src/message_blend/mod.rs]: Yeah, it doesn't need to be at the function level. I fixed it when updating this branch with the new master.[2d7f494](https://github.com/logos-co/nomos-node/pull/905/commits/2d7f4945afb1e1d9e6a398278bb74b5ea24e67f0)

File: nomos-mix/core/src/message_blend/mod.rs [hunk 2]
```diff
@@ -10,41 +10,64 @@ use std::pin::Pin;
...
+    M::PublicKey: Clone + PartialEq,
 {
-    pub fn new(input_stream: S, settings: MessageBlendSettings, rng: Rng) -> Self {
-        let cryptographic_processor = CryptographicProcessor::new(settings.cryptographic_processor);
+    pub fn new(
+        input_stream: S,
+        settings: MessageBlendSettings<M>,
+        membership: Membership<M>,
+        cryptographic_processor_rng: Rng,
+        temporal_processor_rng: Rng,
```
Comment (youngjoon-lee) [inline nomos-mix/core/src/message_blend/mod.rs:61]: We need two Rngs. One for crypto processor, and the other for the temporal processor.

File: nomos-mix/core/src/message_blend/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (youngjoon-lee) [reply nomos-mix/core/src/message_blend/mod.rs]: I requested you for re-review but you can just review the additional commits. I merged the master into this branch "with some changes" that I mentioned below, and removed unnecessary trait bounds.

File: nomos-mix/core/src/message_blend/crypto.rs [hunk 3]
```diff
@@ -1,33 +1,62 @@
...
-pub struct CryptographicProcessor {
-    settings: CryptographicProcessorSettings,
+pub struct CryptographicProcessor<R, M>
+where
+    M: MixMessage,
+{
+    settings: CryptographicProcessorSettings<M::PrivateKey>,
+    membership: Membership<M>,
+    rng: R,
+    _mix_message: PhantomData<M>,
```
Comment (danielSanchezQ) [inline nomos-mix/core/src/message_blend/crypto.rs:16]: leftover?

File: nomos-mix/core/src/message_blend/mod.rs (hunk 2 above)
Comment (danielSanchezQ) [reply nomos-mix/core/src/message_blend/mod.rs:61]: 1. Aditionally we could spawn 2 rngs from a single rng source passed here. But I do not think it is important atm.

**Review by danielSanchezQ — APPROVED** [review]

File: nomos-mix/core/src/message_blend/mod.rs (hunk 2 above)
Comment (youngjoon-lee) [reply nomos-mix/core/src/message_blend/mod.rs:61]: removed the leftover

File: nomos-mix/core/src/message_blend/crypto.rs (hunk 3 above)
Comment (youngjoon-lee) [reply nomos-mix/core/src/message_blend/crypto.rs]: just removed the leftover.

**Pull request merged**